lib/oban/testing.ex

```diff
@@ -236,6 +246,8 @@ defmodule Oban.Testing do
       |> Executor.call()
       |> Map.fetch!(:result)
 
+    Process.delete(:oban_engine)
```

Comment:
Also deleted in `with_testing_mode` but I didn't recognize a better way to make sure it was reliably cleaned up.